### PR TITLE
Don't serialize None values in observer

### PR DESCRIPTION
**Pull request by jerryzh168**
Similar to https://github.com/pytorch/pytorch/pull/32318, we should stop serializing None values since they can't be broadcasted

### CODE REVIEW

File: torch/quantization/observer.py [hunk 1]
```diff
@@ -841,8 +843,9 @@ def calculate_qparams(self):
 
     def _save_to_state_dict(self, destination, prefix, keep_vars):
         super(HistogramObserver, self)._save_to_state_dict(destination, prefix, keep_vars)
-        destination[prefix + 'min_val'] = self.min_val
-        destination[prefix + 'max_val'] = self.max_val
+        if self.min_val is not None and self.max_val is not None:
```
Comment (raghuramank100) [inline torch/quantization/observer.py]: Will this cause an issue with loading from state dict if strict=True? In that case, we will not have entries for min and max val which can trigger an error.

**Comment by kostmo** [comment]
## :pill: CircleCI build failures summary and remediations
As of commit 583fd574:


**None of the build failures appear to be your fault**.

* **1/1** [broken upstream](https://dr.pytorch.org/master-timeline.html?count=1&sha1=53708e21ed0afd6ff4a9281d626612b2e009ff0f&should_suppress_scheduled_builds=true&should_suppress_fully_successful_columns=true&max_columns_suppress_successful=35) at merge base 53708e21 since Jan 16
  <details>
  <summary>
  Please rebase on the <code>viable/strict</code> branch (expand for instructions)
  </summary>
  
  If your commit is newer than `viable/strict`, you can try basing on an older, stable commit:
  ```
  git fetch origin viable/strict
  git rebase --onto viable/strict $(git merge-base origin/master HEAD)
  ```
  If your commit is older than `viable/strict`:
  ```
  git fetch origin viable/strict
  git rebase viable/strict
  ```
  Check out the [recency history](https://dr.pytorch.org/master-viable-commits.html) of this "viable master" tracking branch.
  
  </details>
  


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=583fd57412384922b028d3e9e35cb68d96b2cc48).

### :construction: 1 upstream failure recognized by patterns:
These builds matched patterns, but were probably **caused by upstream breakages**:
* [caffe2_onnx_py3_6_clang7_ubuntu16_04_test](https://dr.pytorch.org/build-details.html?build_id=87488130) from Jan 16



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

File: torch/quantization/observer.py (hunk 1 above)
Comment (jerryzh168) [reply torch/quantization/observer.py]: I think we can make this key optional by excluding them from missing keys

File: torch/quantization/observer.py [hunk 2]
```diff
@@ -335,18 +335,20 @@ def extra_repr(self):
...
+        optional_keys = ['min_val', 'max_val']
         for name in local_state:
             key = prefix + name
             if key in state_dict:
                 setattr(self, name, state_dict.pop(key))
-            elif strict:
+            elif strict and not key in optional_keys:
                 missing_keys.append(key)
         super(MinMaxObserver, self)._load_from_state_dict(state_dict, prefix, local_metadata, strict,
                                                           missing_keys, unexpected_keys, error_msgs)
```
Comment (jerryzh168) [inline torch/quantization/observer.py]: @raghuramank100 these seem to be repeated for all observers can we move them to `_ObserverBase`?

**Comment by jerryzh168** [comment]
@raghuramank100 please take a look again

**Comment by jerryzh168** [comment]
deprecated by https://github.com/pytorch/pytorch/pull/32733

**Pull request closed without merging**